src/code/padmgr.c

```diff
@@ -316,35 +439,43 @@ void PadMgr_HandlePreNMI(PadMgr* padMgr) {
     PadMgr_RumbleReset(padMgr);
 }
 
-void PadMgr_RequestPadData(PadMgr* padMgr, Input* inputs, s32 mode) {
+/**
+ * Fetches the most recently polled inputs from padmgr
+ *
+ * @param padMgr        PadMgr instance
+ * @param[out] inputs   Array of 4 Input to copy inputs into
+ * @param gamePoll      True if polling inputs for the game state at a lower speed than once per VI
+ */
+void PadMgr_RequestPadData(PadMgr* padMgr, Input inputs[MAXCONTROLLERS], s32 gameRequest) {
```

Comment:
gamePoll / gameRequest

also I didn't expect that explanation so I may be back!

Reply:
This is definitely the most mysterious part for me as well so I hope you will give it some extra thought at some point 😄

Reply:
I guess technically commenting that "if this is called once per VI, then passing gameRequest=true or false changes nothing", is correct
(besides the first ever call due to `inputOut` possibly holding garbage in the gameRequest=false path, but let's ignore that)

(note idk if "changes nothing is correct, as this particular piece of code:
```c
buttonDiff = inputOut->prev.button ^ inputOut->cur.button;
inputOut->press.button = inputOut->cur.button & buttonDiff;
inputOut->rel.button = inputOut->prev.button & buttonDiff;
```
idk if it outputs the same thing when run once in `PadMgr_RequestPadData` or several times in `PadMgr_UpdateInputs` with more than one inputs)

----------

But I think this is better explained by just seeing this boolean as a "start game input frame", since it resets the accumulators,
and the use case (it's true only right before gamestate_update()) seems to comfort that

And other calls use false since VIs may have elapsed since, so they don't want to ruin the accumulators (and also they'd want accurate press/rel data since the last time they called).

At least what is sure to me is that `gameRequest=true` should be a single function's job so this argument may be called with that in mind, like `mainCall` or `startInputFrame` idk